Adds a bunch of items to maint, Syndicate contraband included, tweaks probabilities.

FreeStylaLT opened the pull request:
Given the upcoming Space Law change regarding how contraband is handled: http://nanotrasen.se/wiki/index.php/Contraband this was kind of needed to make the crew able to stumble upon some of the C stuff and to prevent forcing people to "RP" that the guy they caught with soap is "not" a traitor.

Obviously, all of the items that aren't soap or the no-slip shoes are rare as fuck. 

Stetchkin can spawn, but only spawns with 1 bullet and cannot be suppressed.

Also modified the list's probabilities to make everything contraband rarer, though also tweaked the most common things like extinguisher and gas mask along with nothing to make them slightly rarer to not go outside of the set value boundaries, to compensate for the new ones.

Rough estimate of distribution: ![img](http://i.imgur.com/IXKm9j7.png)

In 225 loot spawns only 13 were of syndicate origin, 5 were listed as NC - Not Contraband, as in, you can keep these without them getting confiscated according to space law.

3 AI Detectors, 1 Space Suit, 1 smuggler's satchel, 1 champ stamp, 1 voice changer, 1 c4, 1 dart pistol, 2 soaps, 2 clips.

EDIT: This is just how the RNG turned out for this specific batch of 225, point is that all of this additional crap is extremely rare and will not be seen all that often

Current list of new regular additions:

- Duct tape
- brown shoes
- black jumpsuit
- ambrosia deus seeds
- ambrosia vulgaris seeds
- extended oxygen tank

Contraband (rare, some rare as _fuck_) additions:

- Clown grenade
- ambrosia cruciatus seeds
- 1-bullet-in-clip stetchkin
- 10mm magazines
- syndisoap
- dart syringe gun
- uni. suppressor
- chameleon jumpsuit
- chameleon stamp
- No-slip shoes
- voice changer gas mask
- dna scrambler
- smuggler's satchel
- syndicate toolbox
- neutered version of syndicate surgery bag (doesn't contain MMI, suit and gag)
- military belt
- boxed syndi space suit
- AI detector
- Storage implanter
- Syndicate deck of cards
- Briefcase full o' mullah

:cl:FreeStylaLT
add: Added a bunch of new items to maint, some rare uplink gear included. (see PR for full list)
tweak: made gas masks, nothing and extinguishers less common in maint
/:cl:

Fox-McCloud commented on code/modules/projectiles/guns/projectile/pistol.dm:
```diff
@@ -26,6 +26,14 @@
+
+/obj/item/weapon/gun/projectile/automatic/pistol/empty/New()
+		magazine = new /obj/item/ammo_box/magazine/m10mm/empty(src)
+		..()
```
that tabbing.

Fox-McCloud commented on code/game/objects/effects/spawners/lootdrop.dm:
```diff
@@ -58,65 +58,92 @@
+				/obj/item/weapon/grenade/clown_grenade = 5,
+				/obj/item/seeds/ambrosiavulgarisseed/cruciatus = 5,
+				/obj/item/clothing/under/contortionist = 5,
+				/obj/item/weapon/reagent_containers/hypospray/autoinjector/stimulants = 5,
```
nuuuu, nuuuu, this is just a disaster waiting to happen =p

TheDZD replied:
Yeaaaaah, stimulants are OP as fuck, don't put them in maint.

Earthdivine commented:
I said this over Discord but I'll put it here, at a glance, defiantly no C4, I don't know what the Syndicate Syringe Gun is but that sounds like it would just be given to Science to powergame, are the duffels and bags all empty? Tool box is fine, that's just a regular ass toolbox, No binary key.

I also worry that Security will just give this shit to science to get Illegal tech up.

Dumbdumn5 commented:
C4 might be a bit much due to a bit of potential scumbaggery, but aside from that, all fine with it.

theColdflame commented:
:+1: to the idea but there might be some areas that could use ironing out.

'cannot be supressed' means it can't spawn with a suppresor rather than being unable to have a suppressor placed on it, right?

FreeStylaLT commented:
It cannot BE suppressed

theColdflame commented:
Is there a reason behind that? seems pretty random

AceMcLazer commented:
I'd reckon it would be to not allow whomever finds it, traitor or not, to just have a powerful weapon with no downsides. Since you can print 10mm at the lathe, I would think.

Fox-McCloud commented:
I really don't think it's a good idea to be giving regular crewmembers access to powerful syndicate gear on an every-round basis.

Also uh, that's not how the lootspawn table works for the numbers---please read the documentation on it; it details who it works and what numbers should be if you add something new.

FreeStylaLT commented:
I took that into account by reducing Fire Extinguisher's, Gas Mask's and nothing's spawn weight and using that acquired number to distribute it among the new items.

FlattestGuitar commented on code/modules/projectiles/ammunition/magazines.dm:
```diff
@@ -180,6 +180,9 @@
 	max_ammo = 8
 	multiple_sprites = 2
 
+/obj/item/ammo_box/magazine/m10mm/empty		//for maint drops
```
Eh, I'm not sure how the current gun code works, but doesn't this make the magazine just have space for a single bullet? That'd be super wonky.

FreeStylaLT replied:
It does, it then swaps the regular magazine with this one, making the gun only have 1 bullet, but still accept regular 10mm mags if they get made from the protolathe.

FlattestGuitar commented on code/modules/projectiles/guns/projectile/pistol.dm:
```diff
@@ -26,6 +26,14 @@
 
+/obj/item/weapon/gun/projectile/automatic/pistol/empty	//empty stetchshit for maint spawns
+	desc = "A small, easily concealable 10mm handgun. Had a threaded barrel for suppressors."
+	can_suppress = 0
```
Why is it impossible to suppress? Doesn't seem like it's be that much of a difference and to an extent this will enable weird meta.

FreeStylaLT replied:
Felt too powerful for me while making it, I can change that, though, to make it identical to the regular stetchkin.

tigercat2000 replied:
nah let it be suppressable

Bigtimetripod commented:
Holy hell... just imagine running thru a maintenance and come across a damn handgun... I love this, cause still will make me examine every lil piece of gear i can find in maintenance, also will make people visit the maintenance a lot more and as RnD i have a real good chance to discover something i can use to get illegal up.

I support, and ofc will lower metagaming by sec... SYNDISOAP! BANBATON AND PERMA!

Sky-Ping commented:
+1

LittleBigKid2000 commented:
>calling a magazine a clip

_triggered_

Spacemanspark commented:
I'll clip you

Earthdivine commented:
Adrenaline Implants: Probably a no. I know science can apparently make them, but atleast you know they are around. Plus since science can make them, I don't believe it will fulfill it's purpose of "shaking the meta knowledge" very well for how strong an item is.

Stichkin: Pistols are powerful, the revolvers are proof of that. If it wasn't the fact you can print bullets from any autolathe this would be perfectly fine, but this probably would just end up being used by security if confiscated or a validhunting tool more then anything just because the ammo can be printed.

Thermals: Far too strong to have as a random find. This will fuck over any antagonist round they are found, and if the antags find it then they have a massive power boost from it. Just too good.

Binary key: Will ruin any Malf rounds because the RNG says so. Plus history has shown what happens when the crew is given Binary Keys. Probably not a good idea.

C4: Might be ok but people who hold onto bombs usually don't keep them for a good purpose.

Advanced Pinpointer: Oh god no. Fox might get into this a bit more thoroughly but this thing is the strongest thing on there.

TheDZD commented on code/game/objects/effects/spawners/lootdrop.dm:
```diff
@@ -58,65 +58,88 @@
+				/obj/item/weapon/grenade/clown_grenade = 5,
+				/obj/item/seeds/ambrosiavulgarisseed/cruciatus = 5,
+				/obj/item/clothing/under/contortionist = 5,
+				/obj/item/weapon/reagent_containers/hypospray/autoinjector/stimulants = 5,
```
Stimulants are still here, commenting again because Github is hiding the other comments because they're on outdated diffs.

FreeStylaLT replied:
oh missed those, I'll remove them, but I'd prefer to keep up the discussion for the other items so I wouldn't spam commits

FreeStylaLT commented:
I am willing to argue for C4 -- since only one is really _ever_ going to be found, I do not see how it can be truly abused in any real sense, as I cannot think of a single thing that a non-antag can blow up without getting contacted by admins or causing substantial round damage.

As for Stetchkin, it is a lethal weapon, yes, but it is a pretty crappy one. It takes 3 bullets to even break a bone and they have to hit a single limb. Ammo can be made abundantly, yes, but if you get caught with more than just the magazine inside the gun, you get 404'd. 

Security could utilize it, but again, it is so shitty and inventory-reliant for ammunition I doubt they really would use it over lasers. In addition, I do not think sec getting their hands on a single stetchkin once every 5 or more shifts would really make a huge impact, if any at all.

QQCucumber commented:
Helps make the station more dangerous :+1:

Spacemanspark commented:
My only issue with this is the binary key.
It's really horrible when you get subverted and there's that one guy listening in on you, or that desynced cyborg that you don't know about fucks you over.
Even more so worse when the captain finds it and goes "Oh hey I'm going to harass the AI the rest of the round with it, fuck them.

Earthdivine commented:
Binary key got removed two commits ago.

Spacemanspark commented:
That's what I get for using Github on my phone

tigercat2000 commented on code/game/objects/effects/spawners/lootdrop.dm:
```diff
@@ -58,65 +58,90 @@
+				/obj/item/weapon/c4 = 3,
+				/obj/item/device/multitool/ai_detect = 5,
+				/obj/item/weapon/implanter/storage = 1,
+				/obj/item/toy/cards/deck/syndicate = 5,
```
aren't these the ridiculously powerful throwing cards

Spacemanspark replied:
Yes.

Signmark replied:
Hell yes

TheDZD replied:
10 throwforce per card, at 3 throwspeed.

Keilden commented:
Could broken and empty version of stuff be added? Like a empty beecase or those poison bottles. Also add syndiballoon.

FreeStylaLT commented:
I could add those things, but if they get found on you at all, by the new laws you'd get permabrigged. I do not want to add items you cannot legally have on you at all without capital sentencing.

Necaladun commented:
Add all the non outright syndie stuff, make contraband and weapons less likely. In general this all should be rarer too, it just needs the possibility to have an effect. We can increase specific stuff later if needed, or the lot.

LittleBigKid2000 commented:
Add eswords, blood-red hardsuits, bulldog shotguns, medibeams, staffs of healing, bananium, eshields, sleeping carp scrolls, hacked AI modules, syndicate bombs (the drink), syndicate bombs (the explosive), and buttbots to the possible items, and make them all common.

FreeStylaLT commented:
Do you want me to add a dark gygax as well?

LittleBigKid2000 commented:
Yes, especially the syndicate mechs. Oh, and death squad gear, too.

Bigtimetripod commented:
Oh! Lets not forget demon hearts, make pulse rifle most common thing and this might be controversial, but Captains spare id also!

Fox-McCloud commented on code/game/objects/effects/spawners/lootdrop.dm:
```diff
@@ -58,65 +58,91 @@
+				/obj/item/stack/rods{amount = 10} = 80,
+				/obj/item/stack/rods{amount = 23} = 10,
+				/obj/item/stack/rods{amount = 50} = 10,
 				/obj/item/stack/sheet/cardboard = 2,
```
you left this at 2?

TheDZD commented on code/game/objects/effects/spawners/lootdrop.dm:
```diff
@@ -58,65 +58,91 @@
+				////////////////CONTRABAND STUFF//////////////////
+				/obj/item/weapon/grenade/clown_grenade = 5,
+				/obj/item/seeds/ambrosiavulgarisseed/cruciatus = 5,
+				/obj/item/clothing/under/contortionist = 5,
```
I don't think it's a good idea to give people something that lets them ventcrawl around.

TheDZD commented on code/game/objects/effects/spawners/lootdrop.dm:
```diff
@@ -58,65 +58,91 @@
+				/obj/item/weapon/dnascrambler = 3,
+				/obj/item/weapon/storage/backpack/satchel_flat = 5,
+				/obj/item/weapon/storage/toolbox/syndicate = 5,
+				/obj/item/weapon/storage/backpack/duffel/syndie/surgery = 5,
```
This might also be a bad idea, it has an MMI that gives the syndicate lawset when put into a borg.

TheDZD commented on code/game/objects/effects/spawners/lootdrop.dm:
```diff
@@ -58,65 +58,91 @@
+				/obj/item/weapon/storage/backpack/duffel/syndie/surgery = 5,
+				/obj/item/weapon/storage/belt/military = 5,
+				/obj/item/weapon/storage/box/syndie_kit/space = 5,
+				/obj/item/weapon/c4 = 3,
```
Honestly, C4 is something that crewmembers generally don't have any business using, so not sure what the point of putting this here is, considering most circumstances they would use it would become an administrative issue.

FreeStylaLT commented:
Also removed c4 and contortionist's jumpsuit

FlattestGuitar commented:
We bothered. 

[Atrocious list of maintspawns with probabilities](https://docs.google.com/spreadsheets/d/13uav0e4bAZfZ-7JzqHV3jsqtBh7LMS1yOVagNmVr7bA/edit#gid=0)

@theColdflame helped stare at it.

FreeStylaLT commented:
jesus christ you nerds

Well, assuming these are actually correct values, what does everyone think about the probabilities? Should they be lowered? Raised? Should gas mask spawn less than 99.85% of the time? no seriously gas mask started off with a fucking _150_ value in the beginning

SpaceTimeNow commented:
I don't think this is worth the effort. Banning people that metagame might be harsh but you can't fix that via code, because they will allways be one step ahead of you. It's somewhat like hacking in video games, you can't make this game completly metagame proof and the metagaming/powergaming will just shift somewhere else. 

At it's core ss13 is a RP game and even if you manage to make it mostly powergaming proof, then i fear that you will sacrifice a lot of what makes this game special, just to conform to the powergaming crowd.

Btw i think the proposed spawn chances are out of control, i am no math wizard but when i understand the spreadsheet correct, then we would have on average more then 4 or 5 random antag items spawn in maint every round. That seems a little excessive to me. Additionally they should not spawn at round start, because first thing people would do is run into maint.

LittleBigKid2000 commented:
Dude, what if we could play SS13 with SAO-esque VR headsets that could directly modify our memories and skills

Spacemanspark commented:
Then we would become crazed supermutants and the world would end.

CrAzYPiLoT-SS13 commented:
Fireman: "CUT OFF OXYGEN SUPPLY, BLOB DON'T EVOLUATE WITHOUT OXYGEN"

Necaladun commented:
Remind me/everyone:

What are the current chances of the most dangerous stuff to spawn?
What's the chances for everything else?

theColdflame commented:
@Necaladun 
https://docs.google.com/spreadsheets/d/13uav0e4bAZfZ-7JzqHV3jsqtBh7LMS1yOVagNmVr7bA/edit#gid=0

Necaladun commented:
...redid the maths.

There needs to be a few more things that are non-contraband to decrease their chances I think. Adding in another...~80 weight worth of items or increasing the weight of non-contra stuff would do it.

Then halve the contraband numbers again. If my maths is correct, that should get the about balance I want here -Contraband rare but not an impossibility, and impossible for a traitor to go through maint to find just what they need. Even going through maint as a scientist to get some illegal tech shouldn't happen every round. 

Of course, maths, probability, murphy's law, etc. So from then on we just halve or double it until people stop complaining (lol).

FreeStylaLT commented:
Did what @Necaladun asked me to with contraband - reduced all of the probabilities even more. (Couldn't exactly halve a 1, so 1s stayed as 1.)

- Added black gloves to maint
- Added real ins gloves to maint (as rare as contraband)
- Added mini hoe to maint
- Added a botany tray to maint
- you gon' be able to find your own ambrosia farm in maint now
- ain't that a coolio thing

tigercat2000 commented:
@FreeStylaLT there's no reason you can't halve a one
all numbers in byond are floating point
just change it to `0.5`

FreeStylaLT commented:
oh.

uuugh I don't want to re-commit over like 3 digits uuuuuuuuuuuu

we-- we could do that if we feel like its necessary after testing first right

tigercat2000 commented:
well i dunno if pickweight actually works properly for <1 numbers
but there's nothing inherently preventing you from halving a one :tiger:

FlattestGuitar commented:
And there goes my spreadsheet. :c

FreeStylaLT commented on code/game/objects/effects/spawners/lootdrop.dm:
```diff
@@ -61,13 +61,17 @@
+				/obj/item/clothing/gloves/color/fyellow = 10,
+				/obj/item/clothing/gloves/color/yellow = 5,
+				/obj/item/clothing/gloves/color/black = 20,
+				/obj/machinery/portable_atmospherics/hydroponics = 10,
```
yes you can find full trays in maint now

though they spawn wrenched not sure if a problem

Fox-McCloud replied:
Don't think this one is necessary, it's going to be annoying and block areas off, too.

FreeStylaLT commented:
Trays removed

Fox-McCloud commented:
I'm highly skeptical of the stechkin =V

AndriiYukhymchak commented:
who did math in MAINTSPAWNS.xlsx?
I just feel something is wrong there
